### DISCUSSION TITLE
Do we still need configure(experimental=True) with DSPY 2.5.* ?

### DISCUSSION THREAD

**Discussion by denisergashbaev** [stanfordnlp/dspy#1699] [Q&A]
Hello! After switching to DSPY 2.5, is `dspy.configure(experimental=True)` still required to add the newest functionality?

**Comment by aniruddhaadak80** [comment]
From my point of view, this kind of flag should have a very explicit lifecycle because once users upgrade they should not have to guess whether experimental is still unlocking behavior or merely lingering in examples. If 2.5 made the newer path the default, the docs and examples should say so plainly.

Ambiguous migration flags tend to survive much longer than they should and confuse every new user after the transition.

**Comment by stef41** [comment]
No — `dspy.configure(experimental=True)` is no longer needed in DSPy 2.5+.

The `experimental` flag was used in earlier versions (2.0-2.4) to gate new features like:
- The new `Signature` system
- `MIPROv2` optimizer
- Typed predictors
- New adapter system

As of DSPy 2.5, these features are stable and enabled by default. The `experimental` flag is still accepted by `dspy.configure()` for backward compatibility but has no effect — all previously-experimental features are now the standard behavior.

You can safely remove `dspy.configure(experimental=True)` from your code when using DSPy >= 2.5.

```python
# Before (DSPy 2.0-2.4):
dspy.configure(lm=my_lm, experimental=True)

# After (DSPy 2.5+):
dspy.configure(lm=my_lm)
```
